druntime/src/core/internal/newaa.d

```diff
@@ -76,15 +77,19 @@ private size_t mix(size_t h) @safe pure nothrow @nogc
 
 struct Entry(K, V)
 {
-    /*const*/ K key; // this really should be const, but legacy issues.
+    // can make this const, because we aren't really going to use it aside from
+    // construction.
+    const K key;
     V value;
 }
 
 
 // create a binary-compatible AA structure that can be used directly as an
 // associative array.
-AAShell makeAA(K, V)(V[K] src)
+// NOTE: this must only be called during CTFE
+AAShell makeAA(K, V)(V[K] src) @trusted
 {
+    assert(__ctfe, "makeAA Must only be called at compile time");
```

Comment:
Added this because I had to make this `@trusted`. Note that we can pretty much ignore all attributes because this is CTFE only.